prow/git/v2/interactor.go

```diff
@@ -139,10 +140,50 @@ func (i *interactor) IsDirty() (bool, error) {
 
 // Clone clones the repository from a local path.
 func (i *interactor) Clone(from string) error {
+	repoOpts := RepoOpts{
+		SparseCheckoutDirs:         nil,
+		ShareObjectsWithSourceRepo: false,
+	}
+	return i.CloneWithRepoOpts(from, repoOpts)
+}
+
+// CloneWithRepoOpts clones the repository from a local path, but additionally
+// use any repository options (RepoOpts) to customize the clone behavior.
+func (i *interactor) CloneWithRepoOpts(from string, repoOpts RepoOpts) error {
 	i.logger.Infof("Creating a clone of the repo at %s from %s", i.dir, from)
-	if out, err := i.executor.Run("clone", from, i.dir); err != nil {
+	cloneArgs := []string{"clone"}
+
+	if repoOpts.ShareObjectsWithSourceRepo {
+		cloneArgs = append(cloneArgs, "--shared")
+	}
+
+	// Handle sparse checkouts.
+	if repoOpts.SparseCheckoutDirs != nil {
+		cloneArgs = append(cloneArgs, "--sparse")
+	}
+	cloneArgs = append(cloneArgs, from)
+	cloneArgs = append(cloneArgs, i.dir)
```

Comment:
nit: These could be combined.

Reply:
Indeed. I.e., `cloneArgs = append(cloneArgs, []string{from, i.dir}...)`, I think. Will update.

Reply:
Done.